### DISCUSSION TITLE
Expose `onBlur` event

### DISCUSSION THREAD

**Discussion by jtbandes** [sourcelair/xterm.js#5216] [Q&A] [answered]
I would like to clear the selection when the terminal loses focus. I was able to make it work like this:

```ts
    (terminal as any)._core?.onBlur?.(() => {
      terminal.clearSelection();
    });
```

It would be nice to expose this `onBlur` event on the public Terminal interface so people don't have to use private API for it.

(Probably `onFocus` too, I just didn't happen to need that one.)

**Comment by Tyriar** [comment]
I think in the past we didn't add this as you can just add regular dom listeners to `Terminal.element` after calling `Terminal.open`:

https://github.com/xtermjs/xterm.js/blob/41e8ae395937011d6bf6c7cb618b851791aed395/typings/xterm.d.ts#L831-L834

**Comment by jtbandes** [comment]
I tried that as well, but it didn't work.  Example: https://codesandbox.io/p/sandbox/xtermjs-test-forked-rfznf7

**Comment by jtbandes** [comment, marked as answer]
Ah I see, it works with `terminal.textarea` but not `terminal.element`. Thanks! 🎉
